### PR TITLE
Add new exercise

### LINKED ISSUE #1256: Add script to get started with implementing a new exercise
We have a [GitHub template](https://github.com/exercism/csharp/blob/master/.github/ISSUE_TEMPLATE/new_exercise.md) that describes the steps to implement a new exercise. We should create a script to automate these steps where possible.

### PR DESCRIPTION
**Pull request by ErikSchierboom**
Closes #1256

### CODE REVIEW

File: add-new-exercise.ps1 [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+.EXAMPLE
+    The example below will add the "acronym" exercise
+    PS C:\> ./add-new-exercise.ps1 acronym
+.EXAMPLE
+    The example below will add the "acronym" exercise as a core exercise
+    PS C:\> ./add-new-exercise.ps1 acronym -Core
+.EXAMPLE
+    The example below will add the "acronym" exercise as a core exercise, with
+    two topics, a specified difficulty and being unlocked by "two-fer"
+    PS C:\> ./add-new-exercise.ps1 acronym -Core -Topics strings -Difficulty 3 -UnlockedBy two-fer
```
Comment (robkeim) [inline add-new-exercise.ps1]: I only see one topic here, but the description above says there are two topics

File: update-docs.ps1 [hunk 2]
```diff
@@ -1,10 +1,28 @@
...
+    Regenerate the docs.
+.DESCRIPTION
+    Regenerate the docs for all exercises based on the latest canonical data.
+.PARAMETER Exercise
+    The slug of the exercise to regenerate the docs for (optional).
+.EXAMPLE
+    The example below will regenerate all docs
+    PS C:\> ./update-docs.ps1
+.EXAMPLE
+    The example below will regenerate the tests for the "acronym" exercise
```
Comment (robkeim) [inline update-docs.ps1]: regenerate the doc?

File: add-new-exercise.ps1 (hunk 1 above)
Comment (ErikSchierboom) [reply add-new-exercise.ps1]: I'll fix this!

File: update-docs.ps1 (hunk 2 above)
Comment (ErikSchierboom) [reply update-docs.ps1]: Good catch!

**Comment by ErikSchierboom** [comment]
@robkeim I'm currently also working to see if I can easily replace the Cake build script with just plain PowerShell, as I can't get the CI to work. Also, it would remove another dependency, further simplifying the skills needed to work with the build system.

**Comment by ErikSchierboom** [comment]
@robkeim The PR has been updated. Let me know what you think.

File: test.ps1 [hunk 3]
```diff
@@ -22,9 +22,79 @@ param (
...
+    Write-Output "Linting config.json"
+    RunCommand "./bin/fetch-configlet"
+    RunCommand "./bin/configlet lint ."
+}
+
+function BuildGenerators { 
+    Write-Output "Building generators"
+    RunCommand "dotnet build ./generators"
+}
+function Clean {
```
Comment (robkeim) [inline test.ps1]: Nit: this function and the function below don't have an empty newline before them

**Comment by ErikSchierboom** [comment]
Thanks for the review @robkeim!

**Pull request merged**